### DISCUSSION TITLE
Is it possible to use sql lite and postgres in same application?

### DISCUSSION THREAD

**Discussion by tyrsday29** [diesel-rs/diesel#4042] [Q&A]
Looked around and found some conflicting/old answers and was wondering about it. 

Going to be using a managed postgres database for certain things, and want to avoid the cost of lookups for certain things like products that would involve constant lookups... would also be nice to use as a backup for session data and potentially like a cache in certain instances, but I don't want to get greedy. 

Any support for this/way to do it or am I screwed with sled/rocksd?

I'm sure this gets asked a lot, any answers would be appreciated!

**Comment by weiznich** [comment]
You can use different backends in the same application, you just need to enable the feature flags for all the backends you need.

After that it depends a bit on the details what you are trying to do. If you want to use different backends for different parts of your application, you can just have separate modules/crates that work with each backend as you would do with a single backend only. If you want to be able to really swap out the backend for the same code a bit more work is required. You would need to make sure that the schema generated for one database matches the other backend. After that you can use the [`#[derive(MultiConnection)]` macro](https://docs.diesel.rs/2.1.x/diesel/derive.MultiConnection.html) to describe which backends should be supported.
